app/features/receive/use-receive-cashu-token.tsx

```diff
@@ -0,0 +1,358 @@
+  token: Token,
+  mintInfo?: MintInfo,
+  isTestMint?: boolean,
+): CashuAccount => ({
+  type: 'cashu',
+  mintUrl: token.mint,
+  id: '',
+  createdAt: new Date().toISOString(),
+  name: mintInfo?.name ?? 'Unknown Mint',
+  currency: tokenToMoney(token).currency,
+  isTestMint: isTestMint ?? false,
+});
+
+const getClaimableProofs = (
+  unspentProofs: Token['proofs'],
+  cashuPubKey: string,
+) => {
+  let cannotClaimReason: string | undefined;
+  const claimableProofs = unspentProofs.filter((proof) => {
+    if (isPlainSecret(proof.secret)) {
+      return true;
+    }
+    if (!isP2PKSecret(proof.secret)) {
+      cannotClaimReason = 'This ecash contains an unknown spending condition.';
+      return false;
+    }
+    try {
+      const pubkey = getP2PKPubkeyFromProofs([proof]);
+      return pubkey === null || pubkey === cashuPubKey;
+    } catch {
+      console.warn('Failed to get pubkey from proof', proof);
+      return false;
+    }
+  });
+
+  return { claimableProofs, cannotClaimReason };
+};
+
+const getSelectableAccounts = (
+  sourceAccount: CashuAccount,
+  disableCrossMintSwap: boolean,
+  accounts: CashuAccount[],
+) => {
+  return disableCrossMintSwap
+    ? [sourceAccount]
+    : [
+        sourceAccount,
+        ...accounts.filter(
+          (account) =>
+            !account.isTestMint && account.mintUrl !== sourceAccount.mintUrl,
+        ),
+      ];
+};
+
+/**
+ * Get the default account to receive the token.
+ * If the source account is a test mint, then the source account will be returned.
+ * Otherwise it will return the user's default account if it is a cashu account.
+ * Otherwise it will return the source account.
+ * @param selectableAccounts - Accounts available for receiving
+ * @param sourceAccount - Account the token is from
+ * @param disableCrossMintSwap - Whether we can swap to other mints
+ * @param defaultAccount - The user's default account
+ * @returns The default account to receive the token
+ */
+const getDefaultReceiveAccount = (
+  selectableAccounts: CashuAccount[],
+  sourceAccount: CashuAccount,
+  disableCrossMintSwap: boolean,
+  defaultAccount: Account,
+): CashuAccount => {
+  return (
+    selectableAccounts.find(
+      (a) =>
+        a.mintUrl ===
+        (disableCrossMintSwap
+          ? sourceAccount.mintUrl
+          : defaultAccount.type === 'cashu' && defaultAccount.mintUrl),
+    ) ?? sourceAccount
+  );
+};
+
+const getBadges = (
+  account: CashuAccount,
+  allAccounts: CashuAccount[],
+  sourceAccount: CashuAccount,
+  defaultAccount: Account,
+) => {
+  const badges: string[] = [];
+
+  if (sourceAccount.isTestMint) {
+    badges.push('Test Mint');
+  }
+  if (account.mintUrl === sourceAccount.mintUrl) {
+    badges.push('Source');
+  }
+  if (
+    defaultAccount.type === 'cashu' &&
+    account.mintUrl === defaultAccount.mintUrl
+  ) {
+    badges.push('Default');
+  }
+  if (!allAccounts.some((a) => a.mintUrl === account.mintUrl)) {
+    badges.push('Unknown');
```

Comment:
I am not sure that word `Unknown` makes it clear that it means untrusted mint. But maybe it does